Paco/src/com/pacoapp/paco/model/ExperimentUtil.java

```diff
@@ -1,153 +1,37 @@
 package com.pacoapp.paco.model;
 
-import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 import java.util.Map;
 
-import org.joda.time.DateTime;
-
-import android.database.Cursor;
-
 public class ExperimentUtil {
   /**
    * 
-   * @param eventsOutputColumns
-   *          This holds all the column names of the tables Events and Outputs
+   * @param eventsColumns
+   *          This holds all the column names of the tables Events
    *          as keys and the associated table name as value
    * @param colNames
    *          List of names that the client has asked for in the query
    * @return Table name that should be used for the given columns. 
-   *         This will have the following three scenarios 
+   *         This will have the following two scenarios 
    *         - If all input columns are from Events table, will return Events table 
-   *         - If all input columns are from Outputs table, will return Outputs table 
-   *         - If all input columns are a mix of the Outputs and Events table or a '*', will
+   *         - If any one of input columns is not from Events table, will
    *         return Eventsoutputs table
    */
-  public static String identifyTablesInvolved(Map<String, String> eventsOutputColumns, List<String> colNames) {
-    String tableIndicator = null;
+  public static String identifyTablesInvolved(Map<String, String> eventsColumns, List<String> colNames) {
+    // This method is not to validate the column names. This just helps identifying if we need to do a join on outputs table.
+    String tableIndicator = ExperimentProvider.EVENTS_TABLE_NAME;
     if (colNames != null && colNames.size() > 0) {
-      // if the input is *, then we do the join and get all the fields in both
-      // events and outputs
-      String firstColName = colNames.get(0).trim().toUpperCase();
-      if (firstColName != null && firstColName.equals("*")) {
-        return ExperimentProvider.EVENTS_OUTPUTS_TABLE_NAME;
-      }
-      String tableName = eventsOutputColumns.get(firstColName);
-      if (tableName != null) {
-        for (String s : colNames) {
-          String crTableName = eventsOutputColumns.get(s.toUpperCase());
-          // once we get a different table, then we need to do a join
-          if (!tableName.equals(crTableName)) {
-            tableIndicator = ExperimentProvider.EVENTS_OUTPUTS_TABLE_NAME;
-            return tableIndicator;
-          }
-        }
-
-        if (tableName.equalsIgnoreCase(ExperimentProvider.EVENTS_TABLE_NAME)) {
-          tableIndicator = ExperimentProvider.EVENTS_TABLE_NAME;
-        } else if (tableName.equalsIgnoreCase(ExperimentProvider.OUTPUTS_TABLE_NAME)) {
-          tableIndicator = ExperimentProvider.OUTPUTS_TABLE_NAME;
+      for (String s : colNames) {
+        String crTableName = eventsColumns.get(s.toUpperCase());
+        // if we do not get a match in Event column names, then we need to do a join
+        if (crTableName == null) {
```

Comment:
Is it possible for the user to pass through a column that exists in neither table? Is it ok if that happens (given the default behavior of doing a join on Output if the column is not in the eventsColumns?) Just checking.